airflow/gcp/hooks/dataproc.py

```diff
@@ -366,126 +204,626 @@ def build(self) -> Dict:
+            retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+            ``retry`` is specified, the timeout applies to each individual attempt.
+        :type timeout: float
+        :param metadata: Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_job_client(location=location)
+        job = client.get_job(
+            project_id=project_id,
+            region=location,
+            job_id=job_id,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata
+        )
+        return job
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def submit_job(
+        self,
+        location: str,
+        job: Union[Dict, Job],
+        project_id: Optional[str] = None,
+        request_id: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None,
+    ) -> Job:
+        """
+        Submits a job to a cluster.
+
+        :param job: The job resource. If a dict is provided,
+            it must be of the same form as the protobuf message Job
+        :type job: Union[Dict, Job]
+        :param project_id: Required. The ID of the Google Cloud Platform project the cluster belongs to.
+        :type project_id: str
+        :param location: Required. The Cloud Dataproc region in which to handle the request.
+        :type location: str
+        :param request_id: Optional. A tag that prevents multiple concurrent workflow instances
+            with the same tag from running. This mitigates risk of concurrent
+            instances started due to retries.
+        :type request_id: str
+        :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+            retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+            ``retry`` is specified, the timeout applies to each individual attempt.
+        :type timeout: float
+        :param metadata: Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_job_client(location=location)
+        return client.submit_job(
+            project_id=project_id,
+            region=location,
+            job=job,
+            request_id=request_id,
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata
+        )
 
     def submit(
         self,
         project_id: str,
         job: Dict,
         region: str = 'global',
-        job_error_states: Optional[Iterable[str]] = None
+        job_error_states: Optional[Iterable[str]] = None  # pylint:disable=unused-argument
```

Comment:
Why this variable is not used?

Reply:
It's here to preserve old function interface.